SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java

```diff
@@ -514,17 +1437,114 @@ public void testDoy() {
         this.testQueryTwice(query, 224);
     }
 
+    // SELECT EXTRACT(TIMEZONE      FROM DATE '2020-08-11');
+    //ERROR:  unit "timezone" not supported for type date
+    //SELECT EXTRACT(TIMEZONE_M    FROM DATE '2020-08-11');
+    //ERROR:  unit "timezone_m" not supported for type date
+    //SELECT EXTRACT(TIMEZONE_H    FROM DATE '2020-08-11');
+    //ERROR:  unit "timezone_h" not supported for type date
+
     @Test(expected = RuntimeException.class)
     // Calcite does not seem to know about DATE_TRUNC
```

Comment:
Can we add our own DATE_TRUNC for this?

Reply:
I've done this for DCM (extending the parser). The annoying part is the build system, but after that it's pretty straightforward. Create an issue and assign it to me for now, i'll get to it in the coming week or two.

Reply:
The main question is whether we are ready to fork calcite. That's a big step.
This should really be a Calcite issue...

Reply:
It's not a fork, that's the cool part. We basically create a folder like this where we add a few extra definitions: https://github.com/vmware/declarative-cluster-management/tree/master/dcm/src/main/codegen/includes